blocks/api/simple-dom.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+export function createSimpleNode( node, filter = Array ) {
+	if ( ! node ) {
+		return null;
+	}
+
+	if ( node.nodeType === 3 ) {
+		return node.nodeValue;
+	}
+
+	if ( node.nodeType !== 1 ) {
+		return null;
+	}
+
+	return filter(
+		node.nodeName.toLowerCase(),
+		Array.from( node.attributes || [] ).reduce( ( acc, { name, value } ) => ( {
+			...acc,
```

Comment:
Micro-optimization / readability: While this reads nicely, it also creates a shallow clone unnecessarily on each iteration, where we'd be fine to mutate the accumulator:

```js
reduce( node.attributes, ( acc, { name, value } ) => ( {
	acc[ name ] = value;
	return acc;
} )
```

Reply:
or use a `forEach` instead of `reduce` if we want to break the semantics of `reduce` which is normally immutable :wink:

```js
forEach( node.attributes, ( { name, value } ) => acc[ name ] = value );
```